draft-ietf-httpbis-safe-method-w-body.xml

```diff
@@ -317,20 +317,25 @@ q=foo&amp;limit=10&amp;sort=-published
       </t>
       <t>
         The cache key for a QUERY request (see <xref target="HTTP-CACHING" section="2"/>) &MUST;
-        incorporate the request content. When doing so, caches &SHOULD; first normalize request
-        content to remove semantically insignificant differences, thereby improving cache
-        efficiency, by:
+        incorporate the request content.
+      </t>
+      <t>
+        Caches are encouraged to remove semantically insignificant differences first, thereby
+        improving cache efficiency.
+      </t>
+      <t>
+        For instance, by
       </t>
       <ul>
-        <li>Removing content encoding(s) (<xref target="HTTP" section="8.4"/>).</li>
-        <li>Normalizing based upon knowledge of format conventions, as indicated by any
+        <li>removing content encoding(s) (<xref target="HTTP" section="8.4"/>).</li>
+        <li>normalizing based upon knowledge of format conventions, as indicated by any
           media subtype suffix in the request's Content-Type field (e.g., "+json", see
           <xref target="RFC6838" section="4.2.8"/>).</li>
-        <li>Normalizing based upon knowledge of the semantics of the content itself, as indicated by
+        <li>normalizing based upon knowledge of the semantics of the content itself, as indicated by
           the request's Content-Type field.</li>
       </ul>
       <t>
-        Note that any such normalization is performed solely for the purpose of generating a cache key; it
+        Note that any such transformation is performed solely for the purpose of generating a cache key; it
```

Comment:
s/is/should be/

Reply:
"is" is a statement of fact.

If we would say "should", we'd have to consider under which circumstances it would be ok to ignore the requirement.

Reply:
"should" not being normative, I disagree, but we can split this by using "would".  "Note that any such transformation _would be_ performed solely for ...".

Reply:
I don't think this changes the meaning of the statement in any way.

Reply:
I don't want to give the impression that this is something that caches should do.  You want exactly that near as I can tell.  Clearly we differ on this.  I think we've reached the end of productive discussion on GitHub issues commentary.